rules/scripts/bitstreams_workspace.py

```diff
@@ -138,13 +141,29 @@ def GetBitstreamsAvailable(self, refresh):
                         f'Bitstream cache missing {self.latest_update}.')
                 sys.exit(1)
             return
-        document = self.Get('').decode('utf-8')
-        et = xml.etree.ElementTree.fromstring(document)
-        for content in et.findall('Contents', XMLNS):
-            for key in content.findall('Key', XMLNS):
-                m = re.search(r'bitstream-([0-9A-Fa-f]+).tar.gz', key.text)
-                if m:
-                    self.available[m.group(1)] = key.text
+
+        # Fetching the list of all entries in the cache may require multiple
+        # requests since GCP will paginate long reponses. Markers are used to
+        # fetch subsequent pages.
+        # See the GCP docs and the XML REST API reference for more details:
+        # https://cloud.google.com/storage/docs/paginate-results
+        # https://cloud.google.com/storage/docs/xml-api/get-bucket-list
+        marker = ''
+        while True:
+            document = self.Get('', marker=marker).decode('utf-8')
+            et = xml.etree.ElementTree.fromstring(document)
+            for content in et.findall('Contents', XMLNS):
+                for key in content.findall('Key', XMLNS):
+                    m = re.search(r'bitstream-([0-9A-Fa-f]+).tar.gz', key.text)
+                    if m:
+                        self.available[m.group(1)] = key.text
+            # Handle any pagination
+            is_truncated_elt = et.find('IsTruncated', XMLNS)
```

Comment:
It would be nice to have a unit test that mocks out the `Get()` method, but I think I'd rather merge this faster because it's already broken!

Reply:
Agreed - I attempted to do some of the mocking last night, and it was proving a bit challenging to return the correct XML payload. I'm happy to revisit this later in a different PR.

Reply:
Fine by me. Happy to help if you're feeling stuck.

Reply:
Addressed in #16631